### PR TITLE
[google_adsense] Adds H5 Games Ads support to package.

### PR DESCRIPTION
**Pull request by astivi**
This PR adds support for the H5 Games Ads (Ad Placement API) product to `package:google_adsense` through a new `h5.dart` library.

Product page:

* https://adsense.google.com/start/h5-games-ads

JS API reference:

* https://developers.google.com/ad-placement/apis

## Issues

* Continuation of: https://github.com/flutter/packages/pull/6871
* Continuation of: https://github.com/flutter/packages/pull/8233
* Part of: https://github.com/flutter/flutter/issues/40376

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by ditman** [comment]
Adding myself to this so I can take a look!

**Comment by astivi** [comment]
> Adding myself to this so I can take a look! @astivi, is this something you want to merge or are still working on it?

Hey there @ditman,
I'll probably join these changes on #6871 after that is merged in! Don't worry about reviewing this for now. Thanks!

**Comment by ditman** [comment]
> I'll probably join these changes on https://github.com/flutter/packages/pull/6871 after that is merged in! Don't worry about reviewing this for now. Thanks!

I was going to suggest that, or at least to share some code (for example the installing of the JS library). Thanks!

**Comment by ditman** [comment]
(Created a [branch](https://github.com/ditman/flutter-packages/tree/add-h5-ads) to incorporate this PR into @sokoloff06's changes before they land in main)

**Comment by ditman** [comment]
(Actually will keep coding on the other branch; may push to this PR after I'm done there!)

**Comment by ditman** [comment]
Received some comments over email from @alaknanda, implementing those!

**Review by sokoloff06 — COMMENTED** [review]
LGTM with minor comments! @ditman great work wrapping it in a developer friendly package 🌟 


Just an idea: would it make sense to move js_interop folder inside of the core since it's for shared functionality? Then we would have core, h5 and adsense top-level folders and each would have relevant js_interop files inside
image

**Review by sokoloff06 — APPROVED** [review]

File: packages/google_adsense/README.md [hunk 1]
```diff
@@ -30,40 +37,199 @@ void main() async {
...
+```
+
+See the Ad Placement API reference on
+[adConfig](https://developers.google.com/ad-placement/apis/adconfig)
+for a full explanation of all the available parameters.
+
+
+## AdSense
+
+The `experimental/ad_unit_widget.dart` library provides an ad unit `Widget` that
```
[the remark this answers is not in the document]
Comment (ditman) [reply packages/google_adsense/README.md]: Yep, I like this better (I did flip-flop on this one quite a little bit haha, thanks for nudging in the right direction!)

**Comment by ditman** [comment]
All right, this should be ready to land now! Applying `autosubmit`!

**Comment by ditman** [comment]
(Whoops, forgot to move the js-interop around, brb)

**Comment by ditman** [comment]
All right, let's try again!

**Pull request merged**